src/services/filterManager.ts

```diff
@@ -301,13 +315,28 @@ export const formatFromBasicSearch = (groups = [[]]) => {
       }
 
       if (filter.operator === 'contains') {
-        formattedFilter.bool.must.push({
-          regexp: { [filter.attribute]: '.*' + filter.value.toLowerCase() + '.*' }
-        })
+        if (mappings[filter.attribute].type === "text") {
+          formattedFilter.bool.must.push({
+            match_phrase_prefix: { [filter.attribute]: filter.value }
+          })
+        }
+        if (mappings[filter.attribute].type === "keyword") {
+          formattedFilter.bool.must.push({
+            regexp: { [filter.attribute]: ".*" + buildCaseInsensitiveRegexp(filter.value) + ".*" }
```

Comment:
In order for the regex to behave like the `match_phrase_prefix` operator...
```suggestion
            regexp: { [filter.attribute]:  buildCaseInsensitiveRegexp(filter.value) + ".*" }
```